File: RetrofitReservation/app/src/main/java/ma/ensa/retrofitreservation/adapters/ChambreAdapter.java
package ma.ensa.retrofitreservation.adapters;

import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.widget.TextView;

import androidx.recyclerview.widget.RecyclerView;

import java.util.List;

import ma.ensa.retrofitreservation.ListChambreActivity;
import ma.ensa.retrofitreservation.R;
import ma.ensa.retrofitreservation.models.Chambre;

public class ChambreAdapter extends RecyclerView.Adapter<ChambreAdapter.ChambreViewHolder> {

    private List<Chambre> chambres;

    public ChambreAdapter(List<Chambre> chambres, ListChambreActivity listChambreActivity) {
        this.chambres = chambres;
    }

    @Override
    public ChambreViewHolder onCreateViewHolder(ViewGroup parent, int viewType) {
        View view = LayoutInflater.from(parent.getContext()).inflate(R.layout.item_chambre, parent, false);
        return new ChambreViewHolder(view);
    }

    @Override
    public void onBindViewHolder(ChambreViewHolder holder, int position) {
        Chambre chambre = chambres.get(position);
        holder.chambreName.setText(chambre.getTypeChambre().toString()); // Affichage du type de chambre
        holder.chambrePrice.setText(String.format("Prix: %.2f", chambre.getPrix())); // Affichage du prix
        holder.chambreAvailability.setText(chambre.getDispoChambre().toString()); // Affichage de la disponibilité
    }

    @Override
    public int getItemCount() {
        return chambres.size();
    }

    public static class ChambreViewHolder extends RecyclerView.ViewHolder {
        TextView chambreName, chambrePrice, chambreAvailability;

        public ChambreViewHolder(View itemView) {
            super(itemView);
            chambreName = itemView.findViewById(R.id.chambre_name);
            chambrePrice = itemView.findViewById(R.id.chambre_price);
            chambreAvailability = itemView.findViewById(R.id.chambre_availability);
        }
    }
}

File: ReservationRetrofit_KOTLIN/app/src/main/java/ma/ensaj/reservationretrofit_kotlin/adapters/ChambreAdapter.kt
package ma.ensaj.reservationretrofit_kotlin.adapters

import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import android.widget.TextView
import androidx.recyclerview.widget.RecyclerView
import ma.ensaj.reservationretrofit_kotlin.ListChambreActivity
import ma.ensaj.reservationretrofit_kotlin.R
import ma.ensaj.reservationretrofit_kotlin.models.Chambre
import java.lang.String

class ChambreAdapter(
    private val chambres: List<Chambre>,
    listChambreActivity: ListChambreActivity
) : RecyclerView.Adapter<ChambreAdapter.ChambreViewHolder>() {

    override fun onCreateViewHolder(parent: ViewGroup, viewType: Int): ChambreViewHolder {
        val view = LayoutInflater.from(parent.context).inflate(R.layout.item_chambre, parent, false)
        return ChambreViewHolder(view)
    }

    override fun onBindViewHolder(holder: ChambreViewHolder, position: Int) {
        val chambre = chambres[position]
        holder.chambreName.text = chambre.typeChambre.toString() // Affichage du type de chambre
        holder.chambrePrice.text = String.format("Prix: %.2f", chambre.prix) // Affichage du prix
        holder.chambreAvailability.text = chambre.dispoChambre.toString() // Affichage de la disponibilité
    }

    override fun getItemCount(): Int {
        return chambres.size
    }

    class ChambreViewHolder(itemView: View) : RecyclerView.ViewHolder(itemView) {
        val chambreName: TextView = itemView.findViewById(R.id.chambre_name)
        val chambrePrice: TextView = itemView.findViewById(R.id.chambre_price)
        val chambreAvailability: TextView = itemView.findViewById(R.id.chambre_availability)
    }
}
